backend/generator.js

```diff
@@ -9,41 +9,114 @@ exports.executeScript = function (socket, formData) {
     socket.emit('err-logs', "Failed to generated documentation due to an error in input fields");
     return false;
   }
-  var email = formData.email;
-  var gitUrl = formData.gitUrl;
-  var docTheme = formData.docTheme;
-  var uniqueId = uuidV4();
-  var webUI = "true"
-
-  var donePercent = 0;
-
-  const args = [
-    "-g", gitUrl,
-    "-t", docTheme,
-    "-m", email,
-    "-u", uniqueId,
-    "-w", webUI
-  ];
-
-  var process = spawn("./generate.sh", args);
-
-  process.stdout.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('logs', {donePercent: (donePercent = donePercent + 4), data: data.toString()});
-  });
-
-  process.stderr.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('err-logs', data.toString());
-  });
-
-  process.on('exit', function (code) {
-    console.log('child process exited with code ' + code);
-    if (code === 0) {
-      socket.emit('success', {email: email, uniqueId: uniqueId, gitUrl: gitUrl});
-    } else {
-      socket.emit('failure', {errorCode: code});
-    }
-  });
-  return true;
+  // logic for one Repository
```

Comment:
@sch00lb0y I know this is still a `WIP`, but I have a concern here. This is not a good design. The webUI shouldn't handle build logic. the purpose of the webUI should be to just take a list of subproject url's and pass them to `generate.sh`. In the event if there is no subproject pass a empty string. Handle the whole build logic in `generate.sh`. That way this would also work with travis out of the box. This is why some more discussion was needed in #160 before starting work on it.